---
title: "Fat_Depots_PrenatalT"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r lib, echo=TRUE, results='hide', message=F, warning=F, include=FALSE}
library(DESeq2)
library(ggplot2)
library(EnhancedVolcano)
library(grid)
library(gridExtra)
library(gridGraphics)
library(magrittr)
library(openxlsx)
library(gplots)
library(svglite)

library(knitr)

path <- "F:/Drive/Misc/RNAseq/Sheep_RNAseq/Fat_Depos/"
tissues <- list(CAT='CAT',RAT='RAT',SAT='SAT',VAT='VAT')

```

## Meta data setup and load counts

Load table of counts per gene from featureCount output and meta data. 

```{r files}
#load counts, clean up column names
counts <- read.table(file.path(path,"Data/counts"), header=T)
gene_info <- counts[,1:6]
counts <- counts[,7:ncol(counts)]
names(counts) <- gsub('.*\\STAR.|\\.Aligned.*','',names(counts))
names(counts) <- gsub('\\.','_',names(counts))
rownames(counts) <- gene_info$Geneid

#load meta data
meta <- read.table(file.path(path,"Data/Run_2878_padmanabha_DemuxStats.txt"),header=T)
meta$id <- paste0('Sample_',meta$SampleID)

#treatment factors formating
meta$Description <- as.character(meta$Description)
meta$tissue <- substring(meta$Description,1,3)
meta$sheep <- gsub('[A-Z]','',meta$Description)
meta$treat <- sapply(meta$Description, FUN=function(x) substring(x, nchar(x)))
meta <- meta[,c('id','tissue','sheep','treat')]
meta$group <- paste0(meta$tissue, meta$treat)
meta$sheep <- ifelse(meta$sheep=='98', '098', meta$sheep)

#match order
table(meta$id %in% colnames(counts))
counts <- counts[,match(meta$id, colnames(counts))]

table(meta$tissue, meta$sheep)
table(meta$tissue, meta$treat)

#typo in demuxstats.txt file, Sample_126166 should be sheep 233
meta[meta$id=='Sample_126166','sheep'] <- 233

save(gene_info, counts, meta, file=file.path(path,'Data/count_data.RDA'))
```

## Processing for GEO upload
```{r}
library(tools)
server.path <- "/nfs/turbo/bakulski1/Datasets/Padmanabhan/ruddle.brcf.med.umich.edu/ruddle.brcf.med.umich.edu/ruddle.brcf.med.umich.edu/Run_2878/"

#load meta data
meta <- read.table(file.path(server.path,"Run_2878_padmanabha_DemuxStats.txt"),header=T)
meta$id <- paste0('Sample_',meta$SampleID)

#treatment factors formating
meta$Description <- as.character(meta$Description)
meta$tissue <- substring(meta$Description,1,3)
meta$sheep <- gsub('[A-Z]','',meta$Description)
meta$treat <- sapply(meta$Description, FUN=function(x) substring(x, nchar(x)))
meta <- meta[,c('id','tissue','sheep','treat')]
meta$group <- paste0(meta$tissue, meta$treat)
meta$sheep <- ifelse(meta$sheep=='98', '098', meta$sheep)
meta[meta$id=='Sample_126166','sheep'] <- 233

#get list of fastq files
raw.files <- list.files(file.path(server.path,"padmanabha"), recursive=T)

#sample table
samples <- data.frame(sample_name = meta$id,
                      title = meta$group,
                      source = meta$tissue,
                      orgamism = 'sheep',
                      treatment = meta$treat,
                      molecule = 'total RNA',
                      raw_file = gsub('Sample_.*/','',raw.files))

#raw file table
md5 <- md5sum(file.path(server.path,"padmanabha",raw.files))
names(md5) <- gsub('Sample_.*/','',raw.files)
raw <- data.frame(file_name = gsub('Sample_.*/','',raw.files),
                  checksum = md5,
                  instrument = 'Illumina HiSeq 4000',
                  single = 'single')


write.csv(samples, file='/nfs/turbo/bakulski1/People/johndou/GEO_temp_transfer/samples_tab.csv', row.names=F, quote=F)
write.csv(raw, file='/nfs/turbo/bakulski1/People/johndou/GEO_temp_transfer/raw_tab.csv', row.names=F, quote=F)
```


## Loading into DESeq2

```{r}

load(file.path(path,'Data/count_data.RDA'))

#remove extra VAT?
extra.vat <- meta[meta$tissue=='VAT' & meta$sheep==179, 'id']

meta <- meta[meta$id!=extra.vat,]
counts <- counts[,meta$id]

#make deseq object
deseq.ds <- DESeqDataSetFromMatrix(countData = counts,
                                    colData = meta,
                                    design = ~ treat)
```


## Construct Descriptive Table

Table with columns: id, treatment, number reads, number genes

```{r}
tab1 <- meta[,c("id", "sheep", "tissue", "treat")]

summ <- read.table(file.path(path,"Data/counts.summary"), header=T)

#get number reads
names(summ) <- gsub('.*\\STAR.|\\.Aligned.*','',names(summ))
rownames(summ) <- summ[,1]
summ <- summ[,-1]

reads <- summ[1,]
reads <- reads[tab1$id]
tab1$reads <- as.numeric(reads)

#compute n genes from counts
n_genes <- colSums(counts(deseq.ds)>0)
identical(names(n_genes),tab1$id)
tab1$genes <- n_genes

summary(aov(reads~treat,data=tab1))
summary(aov(genes~treat,data=tab1))

#get ranges
range(tab1$reads)
range(tab1$genes)

#order table and write
tab1$id <- gsub('.*_','',tab1$id)
tab1 <- tab1[order(tab1$tissue, tab1$treat, tab1$sheep),]

write.csv(tab1, file=file.path(path,"Tables/descriptives_table.csv"), row.names=F, quote=F)
```


## PCA Plot


```{r}
#PCA plot
vsd <- vst(deseq.ds)

plotPCA(vsd, "group")
pcs <- plotPCA(vsd, "group", returnData=TRUE)
pcs$treat <- substr(pcs$group, 4, 4)
pcs$tissue <- substr(pcs$group, 1, 3)
pcs$tissue <- ifelse(pcs$tissue=="CAT", "ECAT",
                     ifelse(pcs$tissue=="RAT","PRAT", pcs$tissue))
pca.group <- ggplot(pcs, aes(x=PC1, y=PC2, col=tissue, shape=treat)) +
  theme(axis.text=element_text(size=18),
        axis.title=element_text(size=20),
        legend.title=element_text(size=18),
        legend.text=element_text(size=16),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(size=0.5)) +
  scale_shape_manual(values=c(19, 1))+
  geom_point(size=4.1, stroke=1.1) +
  xlab("PC1: 50% variance") + ylab("PC2: 11% variance") + labs(col='Tissue', shape='Treatment')


pca.group

ggsave(pca.group, file=file.path(path,"Plots/PC_plot.svg"))


isolate_pc <- function(fat){
  deseq.fat <- deseq.ds[,deseq.ds$tissue==fat]
  vsd.fat <- vst(deseq.fat)
  plotPCA(vsd.fat, "group")
}

isolate_pc('VAT')
isolate_pc('SAT')
isolate_pc('CAT')
isolate_pc('RAT')

```

# Fit models

## RUV
```{r ruv}
library(RUVSeq)
library(edgeR)

### RUVr

ds <- deseq.ds
ds$group <- as.factor(ds$group)
design(ds) <- ~ group

set <- newSeqExpressionSet(counts(ds, normalized=F), phenoData=AnnotatedDataFrame(data=as.data.frame(colData(ds))))
y <- DGEList(counts=counts(ds, normalized=F), group=ds$group)
design <- model.matrix(~ds$group)
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)

fit <- glmFit(y, design)
resid <- residuals(fit, type="deviance")
ruv <- RUVr(x=set, cIdx=rownames(set), k=1, residuals=resid)


ds.ruv <- DESeqDataSetFromMatrix(countData = counts(ruv), colData = pData(ruv), 
                                 design = ~ W_1 + group)
```

## Tissue difference models
```{r tissue mods}
#tissue 1 vs all
tissue_mod <- function(tissue){
  ds.fat = ds.ruv
  ds.fat$FAT = ds.fat$tissue==tissue
  names(colData(ds.fat))[which(names(colData(ds.fat))=='FAT')] <- tissue
  
  ds.fat_c <- ds.fat[,ds.fat$treat=='C']
  ds.fat_t <- ds.fat[,ds.fat$treat=='T']
  
  design(ds.fat_c) <- as.formula(paste0('~ W_1 + ',tissue))
  design(ds.fat_t) <- as.formula(paste0('~ W_1 + ',tissue))
  
  ds.fat_c <- DESeq(ds.fat_c)
  ds.fat_t <- DESeq(ds.fat_t)
  
  models <- list(ds.fat_c, ds.fat_t)
  names(models) <- c('C','T')
  
  return(models)
}

ds.tissue <- lapply(tissues, tissue_mod)
```

## Prenatal T models
```{r prenatal T mods}
#T, within cell type
treat_mod <- function(tissue){
  ds.fat <- ds.ruv
  ds.fat$group <- relevel(ds.fat$group, ref=paste0(tissue,'C'))
  ds.fat <- DESeq(ds.fat)
}

ds.treat <- lapply(tissues, treat_mod)
```

## process model tables
```{r mod process}

res.tissue <- lapply(tissues, FUN=function(tissue){
  ds.fat = ds.tissue[[tissue]]
  res.C = results(ds.fat$C, name=paste0(tissue,'TRUE'))
  res.T = results(ds.fat$T, name=paste0(tissue,'TRUE'))
  
  res <- list(res.C, res.T)
  names(res) <- c('C','T')
  
  return(res)
})

res.treat <- lapply(tissues, FUN=function(tissue){
  ds.fat = ds.treat[[tissue]]
  results(ds.fat, name=sprintf("group_%sT_vs_%sC", tissue, tissue))
})


# match up filtering limits?
# filts <- sapply(list(res.CAT,res.RAT,res.SAT,res.VAT), function(x) attr(x,"metadata")$filterThreshold)
# m.filt <- max(filts)
# 
# ds.CAT <- ds.CAT[res.CAT$baseMean > m.filt,]
# res.CAT <- results(ds.CAT, name="group_CATT_vs_CATC", independentFiltering = F)
# ds.RAT <- ds.RAT[res.RAT$baseMean > m.filt,]
# res.RAT <- results(ds.RAT, name="group_RATT_vs_RATC", independentFiltering = F)
# ds.SAT <- ds.SAT[res.SAT$baseMean > m.filt,]
# res.SAT <- results(ds.SAT, name="group_SATT_vs_SATC", independentFiltering = F)
# ds.VAT <- ds.VAT[res.VAT$baseMean > m.filt,]
# res.VAT <- results(ds.VAT, name="group_VATT_vs_VATC", independentFiltering = F)


lapply(res.treat, summary)



### shrinkage of logFC

shrink.tissue <- lapply(tissues, FUN=function(tissue){
  res.fat = res.tissue[[tissue]]
  shrink.C <- lfcShrink(ds.tissue[[tissue]]$C, coef=paste0(tissue,'TRUE'), res=res.tissue[[tissue]]$C, type='apeglm')
  shrink.T <- lfcShrink(ds.tissue[[tissue]]$T, coef=paste0(tissue,'TRUE'), res=res.tissue[[tissue]]$T, type='apeglm')
  
  shrink <- list(shrink.C, shrink.T)
  names(shrink) <- c('C','T')
  
  return(shrink)
})


shrink.treat <- lapply(tissues, FUN=function(tissue){
  res.fat = res.treat[[tissue]]
  shrink <- lfcShrink(ds.treat[[tissue]], coef=sprintf("group_%sT_vs_%sC", tissue, tissue), res=res.treat[[tissue]], type='apeglm')
})


### order and filter
augment.tab <- function(tab, tissue, test=T){
  
  #mean per group
  counts <- counts(ds, normalized=T)
  
  if(test){
    baseMean_T <- rowMeans(counts[,meta$group==paste0(tissue,'T')])
    baseMean_C <- rowMeans(counts[,meta$group==paste0(tissue,'C')])
    tab$baseMean_T <- baseMean_T[rownames(tab)]
    tab$baseMean_C <- baseMean_C[rownames(tab)]
  }else{
    baseMean_tissue <- rowMeans(counts[,meta$tissue==tissue])
    baseMean_other <- rowMeans(counts[,meta$tissue!=tissue])
    tab[,paste0('baseMean_',tissue)] <- baseMean_tissue[rownames(tab)]
    tab$baseMean_other <- baseMean_other[rownames(tab)]
  }
  
  #order/filter
  tab <- tab[!is.na(tab$padj),]
  tab <- tab[order(tab$padj),]
  
  #fold change
  tab$FC <- 2^tab$log2FoldChange
  
  #arrange columns
  if(test){
    tab <- tab[,c('baseMean_T','baseMean_C','FC','log2FoldChange','lfcSE','pvalue','padj')]
  }else{
    tab <- tab[c(paste0('baseMean_',tissue),'baseMean_other','FC','log2FoldChange','lfcSE','pvalue','padj')]
  }
  
  tab
}


#run to get normalization factors for counts matrix
ds <- DESeq(ds)

#fix up results tables
tab.tissue <- lapply(tissues, FUN=function(tissue){
  res <- shrink.tissue[[tissue]]
  
  tab.C <- augment.tab(res$C, tissue, test=F)
  tab.T <- augment.tab(res$T, tissue, test=F)
  
  list(C=tab.C, T=tab.T)
})

tab.treat <- lapply(tissues, FUN=function(tissue){
  res <- shrink.treat[[tissue]]
  tab <- augment.tab(res, tissue)
})


save(tab.tissue, tab.treat, file=file.path(path,'Data/res_shrink.RDA'))
```
## Counting
```{r counting}
load(file=file.path(path,'Data/res_shrink.RDA'))

stats <- function(x){
  print('N Genes:')
  print(nrow(x))
  print('FC:')
  print(table(abs(x$log2FoldChange) > 1))
  print('P:')
  print(table(x$padj < 0.05))
  print('FC and P:')
  print(table(abs(x$log2FoldChange) > 1 & x$padj < 0.05))
}

stats(tab.tissue$CAT$C)
stats(tab.tissue$RAT$C)
stats(tab.tissue$SAT$C)
stats(tab.tissue$VAT$C)

stats(tab.treat$CAT)
stats(tab.treat$RAT)
stats(tab.treat$SAT)
stats(tab.treat$VAT)
```


## Volcano Plots
```{r volcano}

load(file=file.path(path,'Data/res_shrink.RDA'))


volcano <- function(tab, 
                    select=NULL, 
                    xlim.left.adj=0, xlim.right.adj=0,
                    horiz.lab.adj=0,
                    vert.lab.adj=1.5){
  EnhancedVolcano(tab, 
                title='',
                subtitle='',
                caption='',
                lab=rownames(tab), 
                selectLab=select,
                x='log2FoldChange',
                y='padj',
                ylab = bquote(~-Log[10]~adjusted~italic(P)),
                xlim=c(min(tab[,'log2FoldChange'], na.rm=TRUE)-xlim.left.adj,
                       max(tab[,'log2FoldChange'], na.rm=TRUE)+xlim.right.adj),
                #ylim = c(0,40),
                legendVisible = F,
                pCutoff=0.05,
                FCcutoff=1,
                transcriptLabSize = 5.0,
                transcriptLabFace = 'italic',
                drawConnectors = TRUE,
                transcriptPointSize = 1.6,
                colAlpha = 1,
                transcriptLabhjust = horiz.lab.adj,
                transcriptLabvjust = vert.lab.adj)
}


### cell type diff in control
de.cell.c.cat <- volcano(tab.tissue$CAT$C, select=c('HOXC6','HOXB7','HOXA7','HOXC8','HOXC9','DMRT2','GNAN1','HOXB5','GATA4','MYL7','LOC106991207','AGT','TTN','MYL4','TNNI3'), horiz.lab.adj=0.45)
de.cell.c.rat <- volcano(tab.tissue$RAT$C, select=c('LHX8','HOXC10','ALX1','ADA','HAND2','LOC105608584','CCN5'), xlim.left.adj=1, horiz.lab.adj=0.40)
de.cell.c.sat <- volcano(tab.tissue$SAT$C, select=c('TCF21','WT1','IGF1','IRX2','IRX1','PAX1','ALX3','LOC106991591'), horiz.lab.adj=0.45)
de.cell.c.vat <- volcano(tab.tissue$VAT$C, select=c('EYA1','SHOX2','ISL1','BARX1','GABRB3','NRN1'), horiz.lab.adj=0.3)

ggsave(de.cell.c.cat, file=file.path(path,"Plots/Vol_cell_con_CAT.svg"), width=8, height=5)
ggsave(de.cell.c.rat, file=file.path(path,"Plots/Vol_cell_con_RAT.svg"), width=8, height=5)
ggsave(de.cell.c.sat, file=file.path(path,"Plots/Vol_cell_con_SAT.svg"), width=8, height=5)
ggsave(de.cell.c.vat, file=file.path(path,"Plots/Vol_cell_con_VAT.svg"), width=8, height=5)


### cell type diff in treat
volcano(tab.tissue$CAT$T)
volcano(tab.tissue$RAT$T)
volcano(tab.tissue$SAT$T)
volcano(tab.tissue$VAT$T)



### T-treat
de.treat.cat <- volcano(tab.treat$CAT, select=c('SH3RF2','HOXA6','GPATCH8','ANLN'))
de.treat.rat <- volcano(tab.treat$RAT, select='RHOJ', vert.lab.adj=-0.2)
de.treat.sat <- volcano(tab.treat$SAT, select=c('ELN','MEDAG','RHOJ','CSMD2'), vert.lab.adj=-0.4)
de.treat.vat <- volcano(tab.treat$VAT, select=c('RHOJ','TCEAL2','RAD51AP1','GNG8','LOC106990101','ITGB2','EMC4','C2H9orf40'), vert.lab.adj=-0.4)

ggsave(de.treat.cat, file=file.path(path,"Plots/Vol_treat_CAT.svg"), width=8, height=5)
ggsave(de.treat.rat, file=file.path(path,"Plots/Vol_treat_RAT.svg"), width=8, height=5)
ggsave(de.treat.sat, file=file.path(path,"Plots/Vol_treat_SAT.svg"), width=8, height=5)
ggsave(de.treat.vat, file=file.path(path,"Plots/Vol_treat_VAT.svg"), width=8, height=5)

```



Differential Gene Tables
```{r}
library(openxlsx)
load(file=file.path(path,'Data/res_shrink.RDA'))

write.gene.tab <- function(tab, filename){
  tab$gene <- paste0(' ',rownames(tab))
  tab <- tab[,c(8,1:7)]
  tab[,2] <- round(tab[,2],2)
  tab[,3] <- round(tab[,3],2)
  tab[,4] <- round(tab[,4],4)
  tab[,5] <- round(tab[,5],2)
  tab[,6] <- round(tab[,6],2)
  tab[,7] <- signif(tab[,7],3)
  tab[,8] <- signif(tab[,8],3)
  
  tablist <- list(Order_Pval=tab, Order_FC=tab[order(-abs(tab$log2FoldChange)),])
  
  #write.xlsx(tablist, file.path(path,'Tables',filename), row.names=F)
  
  invisible(tablist$Order_FC)
}

CC.cat <- write.gene.tab(tab.tissue$CAT$C, 'DE cell type T stratified/DE genes CAT cell type controls.xlsx')
CC.rat <- write.gene.tab(tab.tissue$RAT$C, 'DE cell type T stratified/DE genes RAT cell type controls.xlsx')
CC.sat <- write.gene.tab(tab.tissue$SAT$C, 'DE cell type T stratified/DE genes SAT cell type controls.xlsx')
CC.vat <- write.gene.tab(tab.tissue$VAT$C, 'DE cell type T stratified/DE genes VAT cell type controls.xlsx')

write.xlsx(list(ECAT=CC.cat, PRAT=CC.rat, SAT=CC.sat, VAT=CC.vat), file=file.path(path,'Tables/DE genes cell type controls.xlsx'))


CT.cat <- write.gene.tab(tab.tissue$CAT$T, 'DE cell type T stratified/DE genes CAT cell type treat.xlsx')
CT.rat <- write.gene.tab(tab.tissue$RAT$T, 'DE cell type T stratified/DE genes RAT cell type treat.xlsx')
CT.sat <- write.gene.tab(tab.tissue$SAT$T, 'DE cell type T stratified/DE genes SAT cell type treat.xlsx')
CT.vat <- write.gene.tab(tab.tissue$VAT$T, 'DE cell type T stratified/DE genes VAT cell type treat.xlsx')

write.xlsx(list(ECAT=CT.cat, PRAT=CT.rat, SAT=CT.sat, VAT=CT.vat), file=file.path(path,'Tables/DE genes cell type T treat.xlsx'))


T.cat <- write.gene.tab(tab.treat$CAT, 'DE genes CAT T Treat.xlsx')
T.rat <- write.gene.tab(tab.treat$RAT, 'DE genes RAT T Treat.xlsx')
T.sat <- write.gene.tab(tab.treat$SAT, 'DE genes SAT T Treat.xlsx')
T.vat <- write.gene.tab(tab.treat$VAT, 'DE genes VAT T Treat.xlsx')

write.xlsx(list(ECAT=T.cat, PRAT=T.rat, SAT=T.sat, VAT=T.vat), file=file.path(path,'Tables/DE genes prenatal T treat.xlsx'))


### common across tissues
top.cat <- T.cat[abs(T.cat$log2FoldChange)>1 & T.cat$padj<0.05,]
top.rat <- T.rat[abs(T.rat$log2FoldChange)>1 & T.rat$padj<0.05,]
top.sat <- T.sat[abs(T.sat$log2FoldChange)>1 & T.sat$padj<0.05,]
top.vat <- T.vat[abs(T.vat$log2FoldChange)>1 & T.vat$padj<0.05,]

#count up those top in multiple tissues
top.cross <- data.frame(gene=unique(c(rownames(top.vat),rownames(top.sat),rownames(top.rat),rownames(top.cat))))
top.cross$cat <- top.cross$gene %in% rownames(top.cat)
top.cross$rat <- top.cross$gene %in% rownames(top.rat)
top.cross$sat <- top.cross$gene %in% rownames(top.sat)
top.cross$vat <- top.cross$gene %in% rownames(top.vat)
top.cross$sum <- top.cross$cat + top.cross$rat + top.cross$sat + top.cross$vat

#only pick the ones signif in more than one tissue
top.cross <- top.cross[top.cross$sum>1,]
top.cross$gene <- paste0(" ",top.cross$gene)
top.cross$CAT_FC <- T.cat[match(top.cross$gene, T.cat$gene),'FC']
top.cross$CAT_padj <- T.cat[match(top.cross$gene, T.cat$gene),'padj']
top.cross$RAT_FC <- T.rat[match(top.cross$gene, T.rat$gene),'FC']
top.cross$RAT_padj <- T.rat[match(top.cross$gene, T.rat$gene),'padj']
top.cross$SAT_FC <- T.sat[match(top.cross$gene, T.sat$gene),'FC']
top.cross$SAT_padj <- T.sat[match(top.cross$gene, T.sat$gene),'padj']
top.cross$VAT_FC <- T.vat[match(top.cross$gene, T.vat$gene),'FC']
top.cross$VAT_padj <- T.vat[match(top.cross$gene, T.vat$gene),'padj']

write.xlsx(top.cross, file=file.path(path,'Tables','DE genes cross adipose.xlsx'))
```


# Cross Compare DE Genes

## Cell type stratified
```{r }
load(file=file.path(path,'Data/res_shrink.RDA'))

tissue.corr <- function(FAT){
  common <- intersect(rownames(tab.tissue[[FAT]]$C), rownames(tab.tissue[[FAT]]$T))
  cor(tab.tissue[[FAT]]$C[common,'log2FoldChange'], tab.tissue[[FAT]]$T[common,'log2FoldChange'])
}

sapply(tissues, tissue.corr)
```

## Prenatal T effects

Visualizing overlap
```{r}
library(eulerr)
library(UpSetR)

load(file=file.path(path,'Data/res_shrink.RDA'))

get.top <- function(tab){
  rownames(tab[abs(tab$log2FoldChange)>1 & tab$padj<0.05,])
}

top.FAT <- lapply(tab.treat, get.top)
names(top.FAT) <- c("ECAT","PRAT","SAT","VAT")

### upset plot

svg(file.path(path,'Plots/upset_prenatalT.svg'))
upset(fromList(top.FAT),
      mainbar.y.label = "Number\nof Genes\n\n", 
      order.by='freq',
      keep.order=T,
      text.scale=c(2.9, 1.8, 1.8, 1.7, 1.6, 2.2))
dev.off()


```



Table of genes unique to tissue type
```{r}
library(openxlsx)

tables <- list(CAT=tab.treat$CAT,
               RAT=tab.treat$RAT,
               SAT=tab.treat$SAT,
               VAT=tab.treat$VAT)

grab.unique <- function(fat1, fat2, fat3, fat4){
  fat1[!(fat1 %in% c(fat2, fat3, fat4))]
}

uni.genes <- list()
uni.genes$CAT <- grab.unique(top.FAT$CAT, top.FAT$RAT, top.FAT$SAT, top.FAT$VAT)
uni.genes$RAT <- grab.unique(top.FAT$RAT, top.FAT$CAT, top.FAT$SAT, top.FAT$VAT)
uni.genes$SAT <- grab.unique(top.FAT$SAT, top.FAT$CAT, top.FAT$RAT, top.FAT$VAT)
uni.genes$VAT <- grab.unique(top.FAT$VAT, top.FAT$CAT, top.FAT$RAT, top.FAT$SAT)

top.genes <- list(CAT=top.FAT$CAT, RAT=top.FAT$RAT, SAT=top.FAT$SAT, VAT=top.FAT$VAT)

#grab results from table
uni.gene.table <- lapply(names(uni.genes), FUN=function(tissue){
  tab <- tables[[tissue]]
  tab[uni.genes[[tissue]],]
})
names(uni.gene.table) <- names(uni.genes)

top.gene.table <- lapply(names(top.genes), FUN=function(tissue){
  tab <- tables[[tissue]]
  tab[top.genes[[tissue]],]
})
names(top.gene.table) <- names(top.genes)


#format tables
uni.gene.table <- lapply(uni.gene.table, FUN=function(tab){
  if(nrow(tab)>0){
    tab$gene = paste0(' ', rownames(tab))
    tab = tab[,c(8,1,2,3,4,5,6,7)]
    tab = tab[order(-abs(tab$log2FoldChange)),]
  }else{
    'none'
  }
})

top.gene.table <- lapply(top.gene.table, FUN=function(tab){
  if(nrow(tab)>0){
    tab$gene = paste0(' ', rownames(tab))
    tab = tab[,c(8,1,2,3,4,5,6,7)]
    tab = tab[order(-abs(tab$log2FoldChange)),]
  }else{
    'none'
  }
})


write.xlsx(uni.gene.table, file=file.path(path, 'Tables/DE genes prenatal T tissue unique.xlsx'))
write.xlsx(top.gene.table, file=file.path(path, 'Tables/DE genes prenatal T tissue top genes.xlsx'))

```


# LR path

```{r LR path prep}
### RNA Enrich tables
library(org.Hs.eg.db)

#load human ortholog table
ortho <- read.table(file.path(path,"..","orthologs.txt"),sep="\t", header=T, stringsAsFactors = F)

#get normalized counts
norm.counts <- counts(ds, normalized=T)
identical(meta$id, colnames(norm.counts))

#make tables for LR path
#tab = table from DESeq2 results
#cut = what group(s) to include in average counts
#file = desired name of file
LRprep <- function(tab, cut=c('CATC','CATT','RATC','RATT','SATC','SATT','VATC','VATT'), file){
  cut.counts <- norm.counts[,meta$group %in% cut]
  tab.count <- cut.counts[match(rownames(tab),rownames(cut.counts)),]
  tab.ortho <- ortho[match(rownames(tab),ortho$Sheep.gene.name),"Gene.name"]
  tab.ortho <- unlist(mapIds(org.Hs.eg.db, tab.ortho, "ENTREZID", "SYMBOL"))
  
  LR <- data.frame(geneid=tab.ortho[match(rownames(tab),names(tab.ortho))],
                   PValue=tab$pvalue,
                   logFC=tab$log2FoldChange,
                   norm_avg_readcount=rowMeans(tab.count))
  LR <- LR[!is.na(LR$geneid),]
  
  write.table(LR, sep="\t", row.names=FALSE, quote=FALSE, file=file.path(path,"Tables/LRpath/",file))
  
  invisible(LR)
}

LR.CC <- LRprep(tab.tissue$CAT$C, file='CAT_con_cell_type_LR.txt')
LR.RC <- LRprep(tab.tissue$RAT$C, file='RAT_con_cell_type_LR.txt')
LR.SC <- LRprep(tab.tissue$SAT$C, file='SAT_con_cell_type_LR.txt')
LR.VC <- LRprep(tab.tissue$VAT$C, file='VAT_con_cell_type_LR.txt')

LR.CT <- LRprep(tab.tissue$CAT$T, file='CAT_trt_cell_type_LR.txt')
LR.RT <- LRprep(tab.tissue$RAT$T, file='RAT_trt_cell_type_LR.txt')
LR.ST <- LRprep(tab.tissue$SAT$T, file='SAT_trt_cell_type_LR.txt')
LR.VT <- LRprep(tab.tissue$VAT$T, file='VAT_trt_cell_type_LR.txt')

LR.CAT <- LRprep(tab.treat$CAT, file='CAT_T_treat_LR.txt')
LR.RAT <- LRprep(tab.treat$RAT, file='RAT_T_treat_LR.txt')
LR.SAT <- LRprep(tab.treat$SAT, file='SAT_T_treat_LR.txt')
LR.VAT <- LRprep(tab.treat$VAT, file='VAT_T_treat_LR.txt')

```


Pathway Results from LRpath
```{r lrpath}
library(openxlsx)


### read in prenatal T LR results
path.prenatalT <- lapply(tissues, FUN=function(FAT){
  read.table(paste0(path,'Tables/LRpath/',FAT,'_prenatal_T.txt'), sep='\t', header=T, comment.char='%',quote="")
})

### read in cell type difference LR results
path.tissuecon <- lapply(tissues, FUN=function(FAT){
  read.table(paste0(path,'Tables/LRpath/',FAT,'_celltype_controls.txt'), sep='\t', header=T, comment.char='%',quote="")
})


### formatting tables
lr.process <- function(tab){
  names(tab) <- c('id','name','type','n_genes','coeff','OR','pvalue','FDR','direction','Genes_p_05')
  tab <- tab[order(tab$'FDR'),]
  tab
}

path.prenatalT <- lapply(path.prenatalT, lr.process)
path.tissuecon <- lapply(path.tissuecon, lr.process)


### write formatted tables
lapply(tissues, FUN=function(FAT){
  write.csv(path.prenatalT[[FAT]], file=paste0(path,'Tables/RNA-Enrich ',FAT,' T Treat.csv'), row.names=F)
})
write.xlsx(path.prenatalT, file=paste0(path,'Tables/RNA-Enrich T Treat.xlsx'))

lapply(tissues, FUN=function(FAT){
  write.csv(path.tissuecon[[FAT]], file=paste0(path,'Tables/RNA-Enrich ',FAT,' cell type in controls.csv'), row.names=F)
})
write.xlsx(path.tissuecon, file=paste0(path,'Tables/RNA-Enrich cell type in controls.xlsx'))

```

# Pathway Compare

## Heatmaps
```{r path_heat}

path.prenatalT <- lapply(tissues, FUN=function(FAT){
  read.csv(file=paste0(path,'Tables/RNA-Enrich ',FAT,' T Treat.csv'), header=T)
})


# FDR < 0.01
top.paths <- lapply(path.prenatalT,
                    FUN=function(results){
                      results[results$FDR<0.01,]
                    })

# how many with FDR < 0.01 for each?
sapply(top.paths, nrow)

# upset of pathways in common
library(UpSetR)

top.paths.names <- lapply(top.paths, FUN=function(tab){
  tab$id
})

svg(file.path(path,'Plots/upset_pathway.svg'))
upset(fromList(top.paths.names),
      mainbar.y.label = "Number\nof Pathways\n\n", 
      order.by='freq',
      keep.order=T,
      text.scale=c(2.9, 1.8, 1.8, 1.7, 1.6, 2.2))
dev.off()


# make matrix of odds ratios for the pathways
construct_mat <- function(path.set){
  CAT_paths <- path.prenatalT$CAT[match(path.set,path.prenatalT$CAT$id),]
  RAT_paths <- path.prenatalT$RAT[match(path.set,path.prenatalT$RAT$id),]
  SAT_paths <- path.prenatalT$SAT[match(path.set,path.prenatalT$SAT$id),]
  VAT_paths <- path.prenatalT$VAT[match(path.set,path.prenatalT$VAT$id),]
  
  for_clust <- data.frame(CAT=CAT_paths$OR,
                          RAT=RAT_paths$OR,
                          SAT=SAT_paths$OR,
                          VAT=VAT_paths$OR)
  for_clust <- as.matrix(for_clust)
  rownames(for_clust) <- CAT_paths$name
  return(for_clust)
}

top.paths.collapsed <- lapply(top.paths, 
                              FUN=function(X){
                                X$id
                              }) %>%
                        unlist() %>%
                        as.character() %>%
                        unique() 

OR_matrix <- construct_mat(top.paths.collapsed)


# heatmap

color.gradient <- colorRampPalette(c('firebrick4','firebrick3','gainsboro','royalblue3','royalblue4'))
col.breaks <- c(seq(0,1,length=6),seq(1.2,2.1,length=5))
#col.breaks <- c(0,0.66,1,1.33,2.33,4)

png(file.path(path,'Plots/heatmap_paths.png'), res=300, type='cairo', height=6, width=5, units='in')
path_heat <- heatmap.2(OR_matrix
          ,dendrogram = 'none'
          ,trace='none'
          ,density.info='none'
          ,col=color.gradient(10)
          ,breaks=col.breaks
          ,lmat=rbind(c(3,4,0),
                      c(2,1,1))
          ,lwid=c(0.1,0.5,0.1)
          ,lhei=c(0.1,0.4)
          ,labCol=c("ECAT", "PRAT", "SAT", "VAT")
          ,key.title="Odds Ratio"
          ,key.xlab="Odds Ratio"
          ,labRow="")
dev.off()

svglite(file.path(path,'Plots/heatmap_paths.svg'))
heatmap.2(OR_matrix
          ,dendrogram='none'
          ,density.info='none'
          ,trace='none'
          ,col=color.gradient(10)
          ,breaks=col.breaks
          ,margins=c(0,0)
          # ,lmat=rbind(c(0,0,4,0),
          #             c(0,5,5,0),
          #             c(3,1,2,0),
          #             c(0,0,0,0))
          # ,lhei=c(0.2,1.2,4,1)
          # ,lwid=c(1,0.5,3,1.2)
          ,cexCol=2.1
          ,key.xlab="Enrichment Odds Ratio"
          ,key.title="OR"
          #,RowSideColors = ifelse(paths %in% common, "gray60", ifelse(paths %in% LRtop.gc$Id, "gray92", "gray2"))
          #,labRow=NA
          ,labCol=c("ECAT", "PRAT", "SAT", "VAT")
          ,srtCol=0
          #,adjCol=c(0.5,0.8)
          ,sepwidth = c(0,0)
          ,sepcolor = rgb(0,0,0,0,maxColorValue = 255)
          )
#par(xpd=TRUE)
#legend(-0.1,0.5, legend=c("GC","Both","TC"),fill=c("gray92","gray60","gray2"), bty='n')
dev.off()


#write excel file of pathways in heatmap
OR_excel <- OR_matrix[path_heat$rowInd, path_heat$colInd]
OR_excel <- cbind(rownames(OR_excel), OR_excel)
colnames(OR_excel)[1] <- ''
OR_excel <- OR_excel[rev(rownames(OR_excel)),]
OR_excel[,2:5] <- round(as.numeric(OR_excel[,2:5]), 3)
write.xlsx(OR_excel, file=file.path(path,'Tables/Pathway_heatmap_table_values.xlsx'))
```


# Network Diagrams

```{r network}
library(STRINGdb)
library(org.Hs.eg.db)
library(annotate)

ortho <- read.table(file.path(path,"..","orthologs.txt"),sep="\t", header=T, stringsAsFactors = F)

string_db <- STRINGdb$new(species=9606)

#load RNA enrich pathways
path.prenatalT <- lapply(tissues, FUN=function(FAT){
  read.table(paste0(path,'Tables/LRpath/',FAT,'_prenatal_T.txt'), sep='\t', header=T, comment.char='%',quote="")
})

path.tissuecon <- lapply(tissues, FUN=function(FAT){
  read.table(paste0(path,'Tables/LRpath/',FAT,'_celltype_controls.txt'), sep='\t', header=T, comment.char='%',quote="")
})

load(file=file.path(path,'Data/res_shrink.RDA'))

### RAT Chromatin related
chrm.rat <- path.prenatalT$RAT[path.prenatalT$RAT$Name %in% c("intracellular ribonucleoprotein complex", "ribonucleoprotein complex", "nucleolus", "ribosome biogenesis", "ribonucleoprotein complex biogenesis", "rRNA metabolic process", "posttranscriptional regulation of gene expression"),]
chrm.rat <- paste(chrm.rat$SigGenes, collapse=', ')
chrm.rat <- unlist(strsplit(chrm.rat, ', '))
chrm.rat <- trimws(unique(chrm.rat))
chrm.rat.m <- getSYMBOL(chrm.rat, data='org.Hs.eg.db')

chrm.rat.df <- data.frame(gene=chrm.rat.m)
chrm.rat_map <- string_db$map(chrm.rat.df,"gene")
chrm.rat_map <- chrm.rat_map[!is.na(chrm.rat_map$STRING_id),]
chrm.rat_hits <- chrm.rat_map$"STRING_id"
payload_chrm.rat <- string_db$post_payload(chrm.rat_hits, ifelse(tab.treat$RAT[chrm.rat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb RAT chromatin genes.pdf'))
  string_db$plot_network(chrm.rat_hits, add_link=TRUE, required_score=600, payload_id = payload_chrm.rat)
dev.off()

### RAT ECM/cytoskeletal genes
emc.rat <- path.prenatalT$RAT[path.prenatalT$RAT$Name %in% c("proteinaceous extracellular matrix","anchoring junction","adherens junction","myosin complex","actin cytoskeleton","contractile fiber part","contractile fiber","myosin II complex","focal adhesion","myofibril","cell-substrate adherens junction","cell-substrate junction"),]
emc.rat <- paste(emc.rat$SigGenes, collapse=', ')
emc.rat <- unlist(strsplit(emc.rat, ', '))
emc.rat <- trimws(unique(emc.rat))
emc.rat.m <- getSYMBOL(emc.rat, data='org.Hs.eg.db')

emc.rat.df <- data.frame(gene=emc.rat.m)
emc.rat_map <- string_db$map(emc.rat.df,"gene")
emc.rat_map <- emc.rat_map[!is.na(emc.rat_map$STRING_id),]
emc.rat_hits <- emc.rat_map$"STRING_id"
payload_emc.rat <- string_db$post_payload(emc.rat_hits, ifelse(tab.treat$RAT[emc.rat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb RAT ECMcytoskeletal genes.pdf'))
  string_db$plot_network(emc.rat_hits, add_link=TRUE, required_score=600, payload_id = payload_emc.rat)
dev.off()


### CAT ER/Golgi genes
erg.cat <- path.prenatalT$CAT[path.prenatalT$CAT$Name %in% c("ER to Golgi transport vesicle","ER to Golgi transport vesicle membrane","endoplasmic reticulum lumen","oligosaccharyl transferase activity","oligosaccharyltransferase complex","integral component of endoplasmic reticulum membrane","intrinsic component of endoplasmic reticulum membrane","dolichyl-diphosphooligosaccharide-protein glycotransferase activity","Golgi membrane","Protein export","signal peptide processing","endoplasmic reticulum chaperone complex","phagocytic vesicle membrane","vesicle targeting, rough ER to cis-Golgi","COPII vesicle coating","vesicle coating","ER to Golgi vesicle-mediated transport","protein maturation","chaperone-mediated protein folding","vesicle targeting, to, from or within Golgi","endoplasmic reticulum-Golgi intermediate compartment membrane","endoplasmic reticulum-Golgi intermediate compartment"),]
erg.cat <- paste(erg.cat$SigGenes, collapse=', ')
erg.cat <- unlist(strsplit(erg.cat, ', '))
erg.cat <- trimws(unique(erg.cat))
erg.cat.m <- getSYMBOL(erg.cat, data='org.Hs.eg.db')

erg.cat.df <- data.frame(gene=erg.cat.m)
erg.cat_map <- string_db$map(erg.cat.df,"gene")
erg.cat_map <- erg.cat_map[!is.na(erg.cat_map$STRING_id),]
erg.cat_hits <- erg.cat_map$"STRING_id"
payload_erg.cat <- string_db$post_payload(erg.cat_hits, ifelse(tab.treat$CAT[erg.cat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb CAT ER Golgi genes.pdf'))
  string_db$plot_network(erg.cat_hits, add_link=TRUE, required_score=600, payload_id = payload_erg.cat)
dev.off()


### CAT Immune genes
imm.cat <- path.prenatalT$CAT[path.prenatalT$CAT$Name %in% c("antigen processing and presentation of peptide antigen via MHC class I","inflammatory response","antigen processing and presentation of peptide antigen","immune effector process"),]
imm.cat <- paste(imm.cat$SigGenes, collapse=', ')
imm.cat <- unlist(strsplit(imm.cat, ', '))
imm.cat <- trimws(unique(imm.cat))
imm.cat.m <- getSYMBOL(imm.cat, data='org.Hs.eg.db')

imm.cat.df <- data.frame(gene=imm.cat.m)
imm.cat_map <- string_db$map(imm.cat.df,"gene")
imm.cat_map <- imm.cat_map[!is.na(imm.cat_map$STRING_id),]
imm.cat_hits <- imm.cat_map$"STRING_id"
payload_imm.cat <- string_db$post_payload(imm.cat_hits, ifelse(tab.treat$CAT[imm.cat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb CAT Immune genes.pdf'))
  string_db$plot_network(imm.cat_hits, add_link=TRUE, required_score=600, payload_id = payload_imm.cat)
dev.off()


### SAT Chromatin related
chrm.sat <- path.prenatalT$SAT[path.prenatalT$SAT$Name %in% c("histone deacetylase complex","lysine-acetylated histone binding","DNA-methyltransferase activity","nuclear speck","spliceosomal complex","RNA splicing","chromatin modification"),]
chrm.sat <- paste(chrm.sat$SigGenes, collapse=', ')
chrm.sat <- unlist(strsplit(chrm.sat, ', '))
chrm.sat <- trimws(unique(chrm.sat))
chrm.sat.m <- getSYMBOL(chrm.sat, data='org.Hs.eg.db')

chrm.sat.df <- data.frame(gene=chrm.sat.m)
chrm.sat_map <- string_db$map(chrm.sat.df,"gene")
chrm.sat_map <- chrm.sat_map[!is.na(chrm.sat_map$STRING_id),]
chrm.sat_hits <- chrm.sat_map$"STRING_id"
payload_chrm.sat <- string_db$post_payload(chrm.sat_hits, ifelse(tab.treat$SAT[chrm.sat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb SAT chromatin genes.pdf'))
  string_db$plot_network(chrm.sat_hits, add_link=TRUE, required_score=600, payload_id = payload_chrm.sat)
dev.off()


### SAT Cytoskeleton related
skt.sat <- path.prenatalT$SAT[path.prenatalT$SAT$Name %in% c("cilium assembly","cilium organization","microtubule organizing center part","calmodulin binding","positive regulation of GTPase activity","regulation of GTPase activity"),]
skt.sat <- paste(skt.sat$SigGenes, collapse=', ')
skt.sat <- unlist(strsplit(skt.sat, ', '))
skt.sat <- trimws(unique(skt.sat))
skt.sat.m <- getSYMBOL(skt.sat, data='org.Hs.eg.db')

skt.sat.df <- data.frame(gene=skt.sat.m)
skt.sat_map <- string_db$map(skt.sat.df,"gene")
skt.sat_map <- skt.sat_map[!is.na(skt.sat_map$STRING_id),]
skt.sat_hits <- skt.sat_map$"STRING_id"
payload_skt.sat <- string_db$post_payload(skt.sat_hits, ifelse(tab.treat$SAT[skt.sat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb SAT Cytoskeleton genes.pdf'))
  string_db$plot_network(skt.sat_hits, add_link=TRUE, required_score=600, payload_id = payload_skt.sat)
dev.off()


### VAT Chromatin related
chrm.vat <- path.prenatalT$VAT[path.prenatalT$VAT$Name %in% c("chromatin modification", "histone modification", "chromatin", "histone methylation", "regulation of gene expression by genetic imprinting", "nuclear matrix", "integral component of organelle membrane", "nuclear chromatin", "histone lysine methylation", "genetic imprinting", "peptidyl-lysine methylation", "nuclear chromosome", "double-stranded DNA binding", "nuclear chromosome part", "macromolecule methylation", "regulation of translational elongation", "chromatin assembly or disassembly", "chromatin remodeling", "lysine-acetylated histone binding", "protein methylation", "histone deacetylation", "regulation of chromatin organization", "histone deacetylase complex", "sequence-specific double-stranded DNA binding", "transcriptional repressor complex", "DNA modification"),]
#, "histone deubiquitination", "DNA repair", "methylation", "DNA packaging", "RNA polymerase II transcription factor activity, sequence-specific DNA binding", "heterochromatin", "histone-lysine N-methyltransferase activity", "chromatin silencing", "internal peptidyl-lysine acetylation", "histone H3-K9 methylation", "nucleoside transmembrane transport", "peptidyl-lysine acetylation", "nucleosome organization", "regulation of histone methylation", "enhancer sequence-specific DNA binding", "transcription factor activity, protein binding", "DNA-methyltransferase activity", "mRNA metabolic process", "queuine tRNA-ribosyltransferase activity", "protein-lysine N-methyltransferase activity"
chrm.vat <- paste(chrm.vat$SigGenes, collapse=', ')
chrm.vat <- unlist(strsplit(chrm.vat, ', '))
chrm.vat <- trimws(unique(chrm.vat))
chrm.vat.m <- getSYMBOL(chrm.vat, data='org.Hs.eg.db')

chrm.vat.df <- data.frame(gene=chrm.vat.m)
chrm.vat_map <- string_db$map(chrm.vat.df,"gene")
chrm.vat_map <- chrm.vat_map[!is.na(chrm.vat_map$STRING_id),]
chrm.vat_hits <- chrm.vat_map$"STRING_id"
payload_chrm.vat <- string_db$post_payload(chrm.vat_hits, ifelse(tab.treat$VAT[chrm.vat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb VAT chromatin genes.pdf'))
  string_db$plot_network(chrm.vat_hits, add_link=TRUE, required_score=600, payload_id = payload_chrm.vat)
dev.off()


### VAT ER/Golgi related
erg.vat <- path.prenatalT$VAT[path.prenatalT$VAT$Name %in% c("integral component of endoplasmic reticulum membrane","intrinsic component of endoplasmic reticulum membrane","ER membrane protein complex","Golgi membrane","endosome membrane","intrinsic component of Golgi membrane","integral component of Golgi membrane","organellar ribosome","vesicle targeting, to, from or within Golgi","trans-Golgi network transport vesicle membrane"),]
erg.vat <- paste(erg.vat$SigGenes, collapse=', ')
erg.vat <- unlist(strsplit(erg.vat, ', '))
erg.vat <- trimws(unique(erg.vat))
erg.vat.m <- getSYMBOL(erg.vat, data='org.Hs.eg.db')

erg.vat.df <- data.frame(gene=erg.vat.m)
erg.vat_map <- string_db$map(erg.vat.df,"gene")
erg.vat_map <- erg.vat_map[!is.na(erg.vat_map$STRING_id),]
erg.vat_hits <- erg.vat_map$"STRING_id"
payload_erg.vat <- string_db$post_payload(erg.vat_hits, ifelse(tab.treat$VAT[erg.vat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb VAT ER Golgi genes.pdf'))
  string_db$plot_network(erg.vat_hits, add_link=TRUE, required_score=600, payload_id = payload_erg.vat)
dev.off()


### VAT Immune related
imm.vat <- path.prenatalT$VAT[path.prenatalT$VAT$Name %in% c("antigen processing and presentation of exogenous peptide antigen", "antigen processing and presentation of exogenous peptide antigen via MHC class I", "antigen processing and presentation of peptide antigen via MHC class I", "antigen processing and presentation of peptide antigen", "antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent","Fc receptor signaling pathway","immune response-regulating signaling pathway"),]
imm.vat <- paste(imm.vat$SigGenes, collapse=', ')
imm.vat <- unlist(strsplit(imm.vat, ', '))
imm.vat <- trimws(unique(imm.vat))
imm.vat.m <- getSYMBOL(imm.vat, data='org.Hs.eg.db')

imm.vat.df <- data.frame(gene=imm.vat.m)
imm.vat_map <- string_db$map(imm.vat.df,"gene")
imm.vat_map <- imm.vat_map[!is.na(imm.vat_map$STRING_id),]
imm.vat_hits <- imm.vat_map$"STRING_id"
payload_imm.vat <- string_db$post_payload(imm.vat_hits, ifelse(tab.treat$VAT[imm.vat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb VAT Immune genes.pdf'))
  string_db$plot_network(imm.vat_hits, add_link=TRUE, required_score=600, payload_id = payload_imm.vat)
dev.off()


### VAT mitochondrial
mito.vat <- path.prenatalT$VAT[path.prenatalT$VAT$Name %in% c("mitochondrial matrix","organelle inner membrane","mitochondrial inner membrane","mitochondrial membrane","mitochondrial envelope","intrinsic component of organelle membrane","centriole","mitochondrion organization","mitochondrial translational elongation","mitochondrial translation","mitochondrial translational elongation","mitochondrial translation"),]
mito.vat <- paste(mito.vat$SigGenes, collapse=', ')
mito.vat <- unlist(strsplit(mito.vat, ', '))
mito.vat <- trimws(unique(mito.vat))
mito.vat.m <- getSYMBOL(mito.vat, data='org.Hs.eg.db')

mito.vat.df <- data.frame(gene=mito.vat.m)
mito.vat_map <- string_db$map(mito.vat.df,"gene")
mito.vat_map <- mito.vat_map[!is.na(mito.vat_map$STRING_id),]
mito.vat_hits <- mito.vat_map$"STRING_id"
payload_mito.vat <- string_db$post_payload(mito.vat_hits, ifelse(tab.treat$VAT[mito.vat_map$gene,'log2FoldChange']>0, "#0392ff", "#ff1100"))
pdf(file.path(path,'Plots/Network/STRINGdb VAT Mitochondrial genes.pdf'))
  string_db$plot_network(mito.vat_hits, add_link=TRUE, required_score=600, payload_id = payload_mito.vat)
dev.off()



svg(file.path(path,'Plots/Network/legend.svg'))
plot.new()
legend('center',legend=c("Up","Down"), fill=c("#0392ff", "#ff1100"))
dev.off()
```
