---
title: "Fat_Depots_PrenatalT"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r lib, echo=TRUE, results='hide', message=F, warning=F, include=FALSE}
library(DESeq2)
library(ggplot2)
library(EnhancedVolcano)
library(grid)
library(gridExtra)
library(gridGraphics)
library(magrittr)
library(openxlsx)
library(gplots)

library(knitr)

path <- "F:/Drive/Misc/RNAseq/Sheep_RNAseq/Fat_Depos/"
tissues <- list(CAT='CAT',RAT='RAT',SAT='SAT',VAT='VAT')

```

## Meta data setup and load counts

Load table of counts per gene from featureCount output and meta data. 

```{r files}
#load counts, clean up column names
counts <- read.table(file.path(path,"Data/counts"), header=T)
gene_info <- counts[,1:6]
counts <- counts[,7:ncol(counts)]
names(counts) <- gsub('.*\\STAR.|\\.Aligned.*','',names(counts))
names(counts) <- gsub('\\.','_',names(counts))
rownames(counts) <- gene_info$Geneid

#load meta data
meta <- read.table(file.path(path,"Data/Run_2878_padmanabha_DemuxStats.txt"),header=T)
meta$id <- paste0('Sample_',meta$SampleID)

#treatment factors formating
meta$Description <- as.character(meta$Description)
meta$tissue <- substring(meta$Description,1,3)
meta$sheep <- gsub('[A-Z]','',meta$Description)
meta$treat <- sapply(meta$Description, FUN=function(x) substring(x, nchar(x)))
meta <- meta[,c('id','tissue','sheep','treat')]
meta$group <- paste0(meta$tissue, meta$treat)
meta$sheep <- ifelse(meta$sheep=='98', '098', meta$sheep)

#match order
table(meta$id %in% colnames(counts))
counts <- counts[,match(meta$id, colnames(counts))]

table(meta$tissue, meta$sheep)
table(meta$tissue, meta$treat)

#type in demuxstats.txt file, Sample_126166 should be sheep 233
meta[meta$id=='Sample_126166','sheep'] <- 233

save(gene_info, counts, meta, file=file.path(path,'Data/count_data.RDA'))
```


## Loading into DESeq2

```{r}

load(file.path(path,'Data/count_data.RDA'))

#remove extra VAT?
extra.vat <- meta[meta$tissue=='VAT' & meta$sheep==179, 'id']

meta <- meta[meta$id!=extra.vat,]
counts <- counts[,meta$id]

#make deseq object
deseq.ds <- DESeqDataSetFromMatrix(countData = counts,
                                    colData = meta,
                                    design = ~ treat)
```


## Construct Descriptive Table

Table with columns: id, treatment, number reads, number genes

```{r}
tab1 <- meta[,c("id", "treat")]

summ <- read.table(file.path(path,"Data/counts.summary"), header=T)

#get number reads
names(summ) <- gsub('.*\\STAR.|\\.Aligned.*','',names(summ))
rownames(summ) <- summ[,1]
summ <- summ[,-1]

reads <- summ[1,]
reads <- reads[tab1$id]
tab1$reads <- as.numeric(reads)

#compute n genes from counts
n_genes <- colSums(counts(deseq.ds)>0)
identical(names(n_genes),tab1$id)
tab1$genes <- n_genes

summary(aov(reads~treat,data=tab1))
summary(aov(genes~treat,data=tab1))


tab1$id <- gsub('_','-',tab1$id)
write.csv(tab1, file=file.path(path,"Results/descriptives_table.csv"), row.names=F, quote=F)
```


## PCA Plot


```{r}
#PCA plot
vsd <- vst(deseq.ds)

plotPCA(vsd, "group")
pcs <- plotPCA(vsd, "group", returnData=TRUE)
pcs$treat <- substr(pcs$group, 4, 4)
pcs$tissue <- substr(pcs$group, 1, 3)
pca.group <- ggplot(pcs, aes(x=PC1, y=PC2, col=tissue, shape=treat)) +
  theme(axis.text=element_text(size=16),
        axis.title=element_text(size=18),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(size=0.5)) +
  scale_shape_manual(values=c(19, 1))+
  geom_point(size=3.1, stroke=1.1) +
  xlab("PC1: 50% variance") + ylab("PC2: 11% variance")


pca.group

ggsave(pca.group, file=file.path(path,"Results/PC_plot.png"))


isolate_pc <- function(fat){
  deseq.fat <- deseq.ds[,deseq.ds$tissue==fat]
  vsd.fat <- vst(deseq.fat)
  plotPCA(vsd.fat, "group")
}

isolate_pc('VAT')
isolate_pc('SAT')
isolate_pc('CAT')
isolate_pc('RAT')


getPCs = function(object, intgroup="condition", ntop=500)
{
  # calculate the variance for each gene
  rv <- rowVars(assay(object))

  # select the ntop genes by variance
  select <- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]

  # perform a PCA on the data in assay(x) for the selected genes
  pca <- prcomp(t(assay(object)[select,]))

  # the contribution to the total variance for each component
  percentVar <- pca$sdev^2 / sum( pca$sdev^2 )

  if (!all(intgroup %in% names(colData(object)))) {
    stop("the argument 'intgroup' should specify columns of colData(dds)")
  }

  intgroup.df <- as.data.frame(colData(object)[, intgroup, drop=FALSE])
  
  # add the intgroup factors together to create a new grouping factor
  group <- if (length(intgroup) > 1) {
    factor(apply( intgroup.df, 1, paste, collapse=":"))
  } else {
    colData(object)[[intgroup]]
  }

  # assembly the data for the plot
  d <- data.frame(pca$x, group=group, intgroup.df, name=colnames(object))

  attr(d, "percentVar") <- percentVar
  return(d)

}

pcs <- getPCs(vsd, "treat")
ggplot(pcs, aes(x=PC2, y=PC3, col=group)) + 
  #scale_color_discrete(name='Treatment', labels=c('Control','DCVC','DCVC+LPS','LPS')) +
  theme(axis.text=element_text(size=16),
        axis.title=element_text(size=18),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(size=0.5)) +
  geom_point(size=3.1, stroke=1.1) +
  xlab("PC2") + ylab("PC3")

pcs <- getPCs(vsd, "group")
pcs$treat <- substr(pcs$group, 4, 4)
pcs$tissue <- substr(pcs$group, 1, 3)
ggplot(pcs, aes(x=PC2, y=PC3, col=tissue, shape=treat)) + 
  theme(axis.text=element_text(size=16),
        axis.title=element_text(size=18),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(size=0.5)) +
  geom_point(size=3.1, stroke=1.1) +
  scale_shape_manual(values=c(19, 1))+
  xlab("PC2") + ylab("PC3")
```

# Fit models

## RUV
```{r ruv}
library(RUVSeq)
library(edgeR)

### RUVr

ds <- deseq.ds
ds$group <- as.factor(ds$group)
design(ds) <- ~ group

set <- newSeqExpressionSet(counts(ds, normalized=F), phenoData=AnnotatedDataFrame(data=as.data.frame(colData(ds))))
y <- DGEList(counts=counts(ds, normalized=F), group=ds$group)
design <- model.matrix(~ds$group)
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)

fit <- glmFit(y, design)
resid <- residuals(fit, type="deviance")
ruv <- RUVr(x=set, cIdx=rownames(set), k=1, residuals=resid)


ds.ruv <- DESeqDataSetFromMatrix(countData = counts(ruv), colData = pData(ruv), 
                                 design = ~ W_1 + group)
```

## Tissue difference models
```{r tissue mods}
#tissue 1 vs all
tissue_mod <- function(tissue){
  ds.fat = ds.ruv
  ds.fat$FAT = ds.fat$tissue==tissue
  names(colData(ds.fat))[which(names(colData(ds.fat))=='FAT')] <- tissue
  
  ds.fat_c <- ds.fat[,ds.fat$treat=='C']
  ds.fat_t <- ds.fat[,ds.fat$treat=='T']
  
  design(ds.fat_c) <- as.formula(paste0('~ W_1 + ',tissue))
  design(ds.fat_t) <- as.formula(paste0('~ W_1 + ',tissue))
  
  ds.fat_c <- DESeq(ds.fat_c)
  ds.fat_t <- DESeq(ds.fat_t)
  
  models <- list(ds.fat_c, ds.fat_t)
  names(models) <- c('C','T')
  
  return(models)
}

ds.tissue <- lapply(tissues, tissue_mod)
```

## Prenatal T models
```{r prenatal T mods}
#T, within cell type
treat_mod <- function(tissue){
  ds.fat <- ds.ruv
  ds.fat$group <- relevel(ds.fat$group, ref=paste0(tissue,'C'))
  ds.fat <- DESeq(ds.fat)
}

ds.treat <- lapply(tissues, treat_mod)
```

## process model tables
```{r mod process}

res.tissue <- lapply(tissues, FUN=function(tissue){
  ds.fat = ds.tissue[[tissue]]
  res.C = results(ds.fat$C, name=paste0(tissue,'TRUE'))
  res.T = results(ds.fat$T, name=paste0(tissue,'TRUE'))
  
  res <- list(res.C, res.T)
  names(res) <- c('C','T')
  
  return(res)
})

res.treat <- lapply(tissues, FUN=function(tissue){
  ds.fat = ds.treat[[tissue]]
  results(ds.fat, name=sprintf("group_%sT_vs_%sC", tissue, tissue))
})


# match up filtering limits?
# filts <- sapply(list(res.CAT,res.RAT,res.SAT,res.VAT), function(x) attr(x,"metadata")$filterThreshold)
# m.filt <- max(filts)
# 
# ds.CAT <- ds.CAT[res.CAT$baseMean > m.filt,]
# res.CAT <- results(ds.CAT, name="group_CATT_vs_CATC", independentFiltering = F)
# ds.RAT <- ds.RAT[res.RAT$baseMean > m.filt,]
# res.RAT <- results(ds.RAT, name="group_RATT_vs_RATC", independentFiltering = F)
# ds.SAT <- ds.SAT[res.SAT$baseMean > m.filt,]
# res.SAT <- results(ds.SAT, name="group_SATT_vs_SATC", independentFiltering = F)
# ds.VAT <- ds.VAT[res.VAT$baseMean > m.filt,]
# res.VAT <- results(ds.VAT, name="group_VATT_vs_VATC", independentFiltering = F)


lapply(res.treat, summary)



### shrinkage of logFC

shrink.tissue <- lapply(tissues, FUN=function(tissue){
  res.fat = res.tissue[[tissue]]
  shrink.C <- lfcShrink(ds.tissue[[tissue]]$C, coef=paste0(tissue,'TRUE'), res=res.tissue[[tissue]]$C, type='apeglm')
  shrink.T <- lfcShrink(ds.tissue[[tissue]]$T, coef=paste0(tissue,'TRUE'), res=res.tissue[[tissue]]$T, type='apeglm')
  
  shrink <- list(shrink.C, shrink.T)
  names(shrink) <- c('C','T')
  
  return(shrink)
})


shrink.treat <- lapply(tissues, FUN=function(tissue){
  res.fat = res.treat[[tissue]]
  shrink <- lfcShrink(ds.treat[[tissue]], coef=sprintf("group_%sT_vs_%sC", tissue, tissue), res=res.treat[[tissue]], type='apeglm')
})


### order and filter
augment.tab <- function(tab, tissue, test=T){
  
  #mean per group
  counts <- counts(ds, normalized=T)
  
  if(test){
    baseMean_T <- rowMeans(counts[,meta$group==paste0(tissue,'T')])
    baseMean_C <- rowMeans(counts[,meta$group==paste0(tissue,'C')])
    tab$baseMean_T <- baseMean_T[rownames(tab)]
    tab$baseMean_C <- baseMean_C[rownames(tab)]
  }else{
    baseMean_tissue <- rowMeans(counts[,meta$tissue==tissue])
    baseMean_other <- rowMeans(counts[,meta$tissue!=tissue])
    tab[,paste0('baseMean_',tissue)] <- baseMean_tissue[rownames(tab)]
    tab$baseMean_other <- baseMean_other[rownames(tab)]
  }
  
  #order/filter
  tab <- tab[!is.na(tab$padj),]
  tab <- tab[order(tab$padj),]
  
  #fold change
  tab$FC <- 2^tab$log2FoldChange
  
  #arrange columns
  if(test){
    tab <- tab[,c('baseMean_T','baseMean_C','FC','log2FoldChange','lfcSE','pvalue','padj')]
  }else{
    tab <- tab[c(paste0('baseMean_',tissue),'baseMean_other','FC','log2FoldChange','lfcSE','pvalue','padj')]
  }
  
  tab
}


#run to get normalization factors for counts matrix
ds <- DESeq(ds)

#fix up results tables
tab.tissue <- lapply(tissues, FUN=function(tissue){
  res <- shrink.tissue[[tissue]]
  
  tab.C <- augment.tab(res$C, tissue, test=F)
  tab.T <- augment.tab(res$T, tissue, test=F)
  
  list(C=tab.C, T=tab.T)
})

tab.treat <- lapply(tissues, FUN=function(tissue){
  res <- shrink.treat[[tissue]]
  tab <- augment.tab(res, tissue)
})


save(tab.tissue, tab.treat, file=file.path(path,'Data/res_shrink.RDA'))
```


## Volcano Plots
```{r volcano}

load(file=file.path(path,'Data/res_shrink.RDA'))


### cell type control

volcano <- function(tab){
  EnhancedVolcano(tab, 
                title='',
                subtitle='',
                caption='',
                lab=rownames(tab), 
                #selectLab=c('LRRN4','BARX1','TF','EPHX2','HOXC9','PITX2','IRX2','HOXA9'),
                x='log2FoldChange',
                y='padj',
                ylab = bquote(~-Log[10]~adjusted~italic(P)),
                #xlim = c(-4.5,6.0),
                #ylim = c(0,40),
                legendVisible = F,
                pCutoff=0.05,
                FCcutoff=1,
                transcriptLabSize = 3.5,
                #drawConnectors = TRUE,
                transcriptPointSize = 1.8,
                colAlpha = 1,
                transcriptLabhjust = 0.2)
}

### cell type diff in control
volcano(tab.tissue$CAT$C)
volcano(tab.tissue$RAT$C)
volcano(tab.tissue$SAT$C)
volcano(tab.tissue$VAT$C)

### cell type diff in treat
volcano(tab.tissue$CAT$T)
volcano(tab.tissue$RAT$T)
volcano(tab.tissue$SAT$T)
volcano(tab.tissue$VAT$T)




### T-treat
volcano(tab.treat$CAT)
volcano(tab.treat$RAT)
volcano(tab.treat$SAT)
volcano(tab.treat$VAT)



```



Differential Gene Tables
```{r}
library(openxlsx)
load(file=file.path(path,'Data/res_shrink.RDA'))

write.gene.tab <- function(tab, filename){
  tab$gene <- paste0(' ',rownames(tab))
  tab <- tab[,c(8,1:7)]
  tab[,2] <- round(tab[,2],2)
  tab[,3] <- round(tab[,3],2)
  tab[,4] <- round(tab[,4],4)
  tab[,5] <- round(tab[,5],2)
  tab[,6] <- round(tab[,6],2)
  tab[,7] <- signif(tab[,7],3)
  tab[,8] <- signif(tab[,8],3)
  
  tablist <- list(Order_Pval=tab, Order_FC=tab[abs(order(tab$log2FoldChange)),])
  
  write.xlsx(tablist, file.path(path,'Tables',filename), row.names=F)
  
  invisible(tab)
}

CC.cat <- write.gene.tab(tab.tissue$CAT$C, 'DE cell type T stratified/DE genes CAT cell type controls.xlsx')
CC.rat <- write.gene.tab(tab.tissue$RAT$C, 'DE cell type T stratified/DE genes RAT cell type controls.xlsx')
CC.sat <- write.gene.tab(tab.tissue$SAT$C, 'DE cell type T stratified/DE genes SAT cell type controls.xlsx')
CC.vat <- write.gene.tab(tab.tissue$VAT$C, 'DE cell type T stratified/DE genes VAT cell type controls.xlsx')

CT.cat <- write.gene.tab(tab.tissue$CAT$T, 'DE cell type T stratified/DE genes CAT cell type treat.xlsx')
CT.cat <- write.gene.tab(tab.tissue$RAT$T, 'DE cell type T stratified/DE genes RAT cell type treat.xlsx')
CT.cat <- write.gene.tab(tab.tissue$SAT$T, 'DE cell type T stratified/DE genes SAT cell type treat.xlsx')
CT.cat <- write.gene.tab(tab.tissue$VAT$T, 'DE cell type T stratified/DE genes VAT cell type treat.xlsx')

T.cat <- write.gene.tab(tab.treat$CAT, 'DE genes CAT T Treat.xlsx')
T.rat <- write.gene.tab(tab.treat$RAT, 'DE genes RAT T Treat.xlsx')
T.sat <- write.gene.tab(tab.treat$SAT, 'DE genes SAT T Treat.xlsx')
T.vat <- write.gene.tab(tab.treat$VAT, 'DE genes VAT T Treat.xlsx')



### common across tissues
top.cat <- T.cat[abs(T.cat$log2FoldChange)>1 & T.cat$padj<0.05,]
top.rat <- T.rat[abs(T.rat$log2FoldChange)>1 & T.rat$padj<0.05,]
top.sat <- T.sat[abs(T.sat$log2FoldChange)>1 & T.sat$padj<0.05,]
top.vat <- T.vat[abs(T.vat$log2FoldChange)>1 & T.vat$padj<0.05,]

#count up those top in multiple tissues
top.cross <- data.frame(gene=unique(c(rownames(top.vat),rownames(top.sat),rownames(top.rat),rownames(top.cat))))
top.cross$cat <- top.cross$gene %in% rownames(top.cat)
top.cross$rat <- top.cross$gene %in% rownames(top.rat)
top.cross$sat <- top.cross$gene %in% rownames(top.sat)
top.cross$vat <- top.cross$gene %in% rownames(top.vat)
top.cross$sum <- top.cross$cat + top.cross$rat + top.cross$sat + top.cross$vat

#only pick the ones signif in more than one tissue
top.cross <- top.cross[top.cross$sum>1,]
top.cross$gene <- paste0(" ",top.cross$gene)
top.cross$CAT_logFC <- T.cat[match(top.cross$gene, T.cat$gene),'log2FoldChange']
top.cross$CAT_padj <- T.cat[match(top.cross$gene, T.cat$gene),'padj']
top.cross$RAT_logFC <- T.rat[match(top.cross$gene, T.rat$gene),'log2FoldChange']
top.cross$RAT_padj <- T.rat[match(top.cross$gene, T.rat$gene),'padj']
top.cross$SAT_logFC <- T.sat[match(top.cross$gene, T.sat$gene),'log2FoldChange']
top.cross$SAT_padj <- T.sat[match(top.cross$gene, T.sat$gene),'padj']
top.cross$VAT_logFC <- T.vat[match(top.cross$gene, T.vat$gene),'log2FoldChange']
top.cross$VAT_padj <- T.vat[match(top.cross$gene, T.vat$gene),'padj']

write.xlsx(top.cross, file=file.path(path,'Tables','DE genes cross adipose.xlsx'))
```



```{r LR path prep}
### RNA Enrich tables
library(org.Hs.eg.db)

#load human ortholog table
ortho <- read.table(file.path(path,"..","orthologs.txt"),sep="\t", header=T, stringsAsFactors = F)

#get normalized counts
norm.counts <- counts(ds, normalized=T)
identical(meta$id, colnames(norm.counts))

#make tables for LR path
#tab = table from DESeq2 results
#cut = what group(s) to include in average counts
#file = desired name of file
LRprep <- function(tab, cut=c('CATC','CATT','RATC','RATT','SATC','SATT','VATC','VATT'), file){
  cut.counts <- norm.counts[,meta$group %in% cut]
  tab.count <- cut.counts[match(rownames(tab),rownames(cut.counts)),]
  tab.ortho <- ortho[match(rownames(tab),ortho$Sheep.gene.name),"Gene.name"]
  tab.ortho <- unlist(mapIds(org.Hs.eg.db, tab.ortho, "ENTREZID", "SYMBOL"))
  
  LR <- data.frame(geneid=tab.ortho[match(rownames(tab),names(tab.ortho))],
                   PValue=tab$pvalue,
                   logFC=tab$log2FoldChange,
                   norm_avg_readcount=rowMeans(tab.count))
  LR <- LR[!is.na(LR$geneid),]
  
  write.table(LR, sep="\t", row.names=FALSE, quote=FALSE, file=file.path(path,"Tables/LRpath/",file))
  
  invisible(LR)
}

LR.CC <- LRprep(tab.tissue$CAT$C, file='CAT_con_cell_type_LR.txt')
LR.RC <- LRprep(tab.tissue$RAT$C, file='RAT_con_cell_type_LR.txt')
LR.SC <- LRprep(tab.tissue$SAT$C, file='SAT_con_cell_type_LR.txt')
LR.VC <- LRprep(tab.tissue$VAT$C, file='VAT_con_cell_type_LR.txt')

LR.CT <- LRprep(tab.tissue$CAT$T, file='CAT_trt_cell_type_LR.txt')
LR.RT <- LRprep(tab.tissue$RAT$T, file='RAT_trt_cell_type_LR.txt')
LR.ST <- LRprep(tab.tissue$SAT$T, file='SAT_trt_cell_type_LR.txt')
LR.VT <- LRprep(tab.tissue$VAT$T, file='VAT_trt_cell_type_LR.txt')

LR.CAT <- LRprep(tab.treat$CAT, file='CAT_T_treat_LR.txt')
LR.RAT <- LRprep(tab.treat$RAT, file='RAT_T_treat_LR.txt')
LR.SAT <- LRprep(tab.treat$SAT, file='SAT_T_treat_LR.txt')
LR.VAT <- LRprep(tab.treat$VAT, file='VAT_T_treat_LR.txt')

```


Pathway Results from LRpath
```{r lrpath}
library(openxlsx)


### read in prenatal T LR results
path.prenatalT <- lapply(tissues, FUN=function(FAT){
  read.table(paste0(path,'Tables/LRpath/',FAT,'_prenatal_T.txt'), sep='\t', header=T, comment.char='%',quote="")
})

### read in cell type difference LR results
path.tissuecon <- lapply(tissues, FUN=function(FAT){
  read.table(paste0(path,'Tables/LRpath/',FAT,'_celltype_controls.txt'), sep='\t', header=T, comment.char='%',quote="")
})


### formatting tables
lr.process <- function(tab){
  names(tab) <- c('id','name','type','n_genes','coeff','OR','pvalue','FDR','direction','Genes_p_05')
  tab <- tab[order(tab$'pvalue'),]
  tab
}

path.prenatalT <- lapply(path.prenatalT, lr.process)
path.tissuecon <- lapply(path.tissuecon, lr.process)


### write formatted tables
lapply(tissues, FUN=function(FAT){
  write.csv(path.prenatalT[[FAT]], file=paste0(path,'Tables/RNA-Enrich ',FAT,' T Treat.csv'), row.names=F)
})
write.xlsx(path.prenatalT, file=paste0(path,'Tables/RNA-Enrich T Treat.xlsx'))

lapply(tissues, FUN=function(FAT){
  write.csv(path.tissuecon[[FAT]], file=paste0(path,'Tables/RNA-Enrich ',FAT,' cell type in controls.csv'), row.names=F)
})
write.xlsx(path.tissuecon, file=paste0(path,'Tables/RNA-Enrich cell type in controls.xlsx'))

```


# Cross Compare DE Genes

## Cell type stratified
```{r }
load(file=file.path(path,'Data/res_shrink.RDA'))

tissue.corr <- function(FAT){
  common <- intersect(rownames(tab.tissue[[FAT]]$C), rownames(tab.tissue[[FAT]]$T))
  cor(tab.tissue[[FAT]]$C[common,'log2FoldChange'], tab.tissue[[FAT]]$T[common,'log2FoldChange'])
}

sapply(tissues, tissue.corr)
```

## Prenatal T effects

Visualizing overlap
```{r}
library(eulerr)
library(UpSetR)

load(file=file.path(path,'Data/res_shrink.RDA'))

get.top <- function(tab){
  rownames(tab[abs(tab$log2FoldChange)>1 & tab$padj<0.05,])
}

top.FAT <- lapply(tab.treat, get.top)


### venn diagram

vnc <- euler(top.FAT)
p <- plot(vnc, quantities=T, adjust_lables=T, labels=T)

pdf(file.path(path,"/Plots/venn_sig_genes.pdf"), height=10, width=10)
p
dev.off()

vnc


### upset plot

upset(fromList(top.FAT),
      mainbar.y.label = "Number\nof Genes\n\n", 
      order.by='freq',
      keep.order=T,
      text.scale=c(2.3, 1.8, 1.8, 1.7, 1.6, 1.6))

```



Table of genes unique to tissue type
```{r}
library(openxlsx)

tables <- list(CAT=tab.treat$CAT,
               RAT=tab.treat$RAT,
               SAT=tab.treat$SAT,
               VAT=tab.treat$VAT)

grab.unique <- function(fat1, fat2, fat3, fat4){
  fat1[!(fat1 %in% c(fat2, fat3, fat4))]
}

uni.genes <- list()
uni.genes$CAT <- grab.unique(top.FAT$CAT, top.FAT$RAT, top.FAT$SAT, top.FAT$VAT)
uni.genes$RAT <- grab.unique(top.FAT$RAT, top.FAT$CAT, top.FAT$SAT, top.FAT$VAT)
uni.genes$SAT <- grab.unique(top.FAT$SAT, top.FAT$CAT, top.FAT$RAT, top.FAT$VAT)
uni.genes$VAT <- grab.unique(top.FAT$VAT, top.FAT$CAT, top.FAT$RAT, top.FAT$SAT)

#grab results from table
uni.gene.table <- lapply(names(uni.genes), FUN=function(tissue){
  tab <- tables[[tissue]]
  tab[uni.genes[[tissue]],]
})
names(uni.gene.table) <- names(uni.genes)

#format tables
uni.gene.table <- lapply(uni.gene.table, FUN=function(tab){
  if(nrow(tab)>0){
    tab$gene = paste0(' ', rownames(tab))
    tab = tab[,c(8,1,2,3,4,5,6,7)]
    tab = tab[order(-abs(tab$log2FoldChange)),]
  }else{
    'none'
  }
})

write.xlsx(uni.gene.table, file=file.path(path, 'Tables/DE genes prenatal T tissue unique.xlsx'))
```



# Pathway Compare

## Heatmaps
```{r path_heat}

path.prenatalT <- lapply(tissues, FUN=function(FAT){
  read.csv(file=paste0(path,'Tables/RNA-Enrich ',FAT,' T Treat.csv'), header=T)
})


# FDR < 0.01
top.paths <- lapply(path.prenatalT,
                    FUN=function(results){
                      results[results$FDR<0.01,]
                    })

# how many with FDR < 0.01 for each?
sapply(top.paths, nrow)

# upset of pathways in common
library(UpSetR)

top.paths.names <- lapply(top.paths, FUN=function(tab){
  tab$id
})
upset(fromList(top.paths.names),
      mainbar.y.label = "Number\nof Pathways\n\n", 
      order.by='freq',
      keep.order=T,
      text.scale=c(2.3, 1.8, 1.8, 1.5, 1.6, 1.6))



# make matrix of odds ratios for the pathways
construct_mat <- function(path.set){
  CAT_paths <- path.prenatalT$CAT[match(path.set,path.prenatalT$CAT$id),]
  RAT_paths <- path.prenatalT$RAT[match(path.set,path.prenatalT$RAT$id),]
  SAT_paths <- path.prenatalT$SAT[match(path.set,path.prenatalT$SAT$id),]
  VAT_paths <- path.prenatalT$VAT[match(path.set,path.prenatalT$VAT$id),]
  
  for_clust <- data.frame(CAT=CAT_paths$OR,
                          RAT=RAT_paths$OR,
                          SAT=SAT_paths$OR,
                          VAT=VAT_paths$OR)
  for_clust <- as.matrix(for_clust)
  rownames(for_clust) <- CAT_paths$name
  return(for_clust)
}

top.paths.collapsed <- lapply(top.paths, 
                              FUN=function(X){
                                X$id
                              }) %>%
                        unlist() %>%
                        as.character() %>%
                        unique() 

OR_matrix <- construct_mat(top.paths.collapsed)


# heatmap

color.gradient <- colorRampPalette(c('firebrick4','firebrick3','gainsboro','royalblue3','royalblue4'))
col.breaks <- c(seq(0,1,length=6),seq(1.2,2.1,length=5))
#col.breaks <- c(0,0.66,1,1.33,2.33,4)

#svglite(file.path(res,'heatmap_paths.svg'))
path_heat <- heatmap.2(OR_matrix
          ,trace='none'
          ,density.info='none'
          ,col=color.gradient(10)
          ,breaks=col.breaks)

heatmap.2(OR_matrix
          ,dendrogram='none'
          ,density.info='none'
          ,trace='none'
          ,col=color.gradient(10)
          ,breaks=col.breaks
          ,margins=c(0,0)
          # ,lmat=rbind(c(0,0,4,0),
          #             c(0,5,5,0),
          #             c(3,1,2,0),
          #             c(0,0,0,0))
          # ,lhei=c(0.2,1.2,4,1)
          # ,lwid=c(1,0.5,3,1.2)
          ,cexCol=2.1
          ,key.xlab="Enrichment Odds Ratio"
          ,key.title="OR"
          #,RowSideColors = ifelse(paths %in% common, "gray60", ifelse(paths %in% LRtop.gc$Id, "gray92", "gray2"))
          #,labRow=NA
          ,labCol=c("CAT", "RAT", "SAT", "VAT")
          ,srtCol=0
          #,adjCol=c(0.5,0.8)
          ,sepwidth = c(0,0)
          ,sepcolor = rgb(0,0,0,0,maxColorValue = 255)
          )
#par(xpd=TRUE)
#legend(-0.1,0.5, legend=c("GC","Both","TC"),fill=c("gray92","gray60","gray2"), bty='n')
#dev.off()


#write excel file of pathways in heatmap
OR_excel <- OR_matrix[path_heat$rowInd, path_heat$colInd]
OR_excel <- cbind(rownames(OR_excel), OR_excel)
colnames(OR_excel)[1] <- ''
OR_excel <- OR_excel[rev(rownames(OR_excel)),]
OR_excel[,2:5] <- round(as.numeric(OR_excel[,2:5]), 3)
write.xlsx(OR_excel, file=file.path(path,'Tables/Pathway_heatmap_table_values.xlsx'))
```